apps/desktop-wallet/src/components/Amount.tsx

```diff
@@ -54,12 +57,29 @@ interface CustomAmountProps extends AmountBaseProps {
 
 type AmountProps = TokenAmountProps | FiatAmountProps | CustomAmountProps
 
-const Amount = (props: AmountProps) => {
+type AmountPropsWithOptionalAmount =
+  | Optional<TokenAmountProps, 'value'>
+  | Optional<FiatAmountProps, 'value'>
+  | Optional<CustomAmountProps, 'value'>
+
+interface LoaderProps {
+  isLoading?: boolean
+  loaderSizeInPx?: string
+}
```

Comment:
This is the way I found to implement what you suggested in the other PR (moving loader inside Amount component) without loosing the strict types of `TokenAmountProps | FiatAmountProps | CustomAmountProps` where the amount MUST be defined and MUST be of different type for each of the above types.

Reply:
LGTM!